### ISSUE TITLE
argument 1 value '18446744073709551615' exceeds maximum object size 9223372036854775807 [-Werror=alloc-size-larger-than=]

### ISSUE THREAD

**Issue by Mailaender** [Return-To-The-Roots/s25client#679]
The version 0.8.2-16 currently won't compile for @openSUSE Tumbleweed. I assume this is due to a warning (treated as error) in a GCC version you haven't tested yet.

```
[   58s] /home/abuild/rpmbuild/BUILD/s25rttr-0.8.2/s-c/resample-1.8.1/sndlib-20/sound.c: In function 'mus_sound_maxamps':
[   58s] /home/abuild/rpmbuild/BUILD/s25rttr-0.8.2/s-c/resample-1.8.1/sndlib-20/sound.c:914:9: error: argument 1 value '18446744073709551615' exceeds maximum object size 9223372036854775807 [-Werror=alloc-size-larger-than=]
[   58s]    ibufs = (mus_sample_t **)CALLOC(ichans, sizeof(mus_sample_t *));
[   58s] In file included from /home/abuild/rpmbuild/BUILD/s25rttr-0.8.2/s-c/resample-1.8.1/sndlib-20/sound.c:12:0:
[   58s] /usr/include/stdlib.h:445:14: note: in a call to allocation function 'calloc' declared here
[   58s]  extern void *calloc (size_t __nmemb, size_t __size)
[   58s]               ^~~~~~
[   58s] /home/abuild/rpmbuild/BUILD/s25rttr-0.8.2/s-c/resample-1.8.1/sndlib-20/sound.c:918:8: error: argument 1 value '18446744073709551615' exceeds maximum object size 9223372036854775807 [-Werror=alloc-size-larger-than=]
[   58s]    time = (off_t *)CALLOC(ichans, sizeof(off_t));
[   58s] In file included from /home/abuild/rpmbuild/BUILD/s25rttr-0.8.2/s-c/resample-1.8.1/sndlib-20/sound.c:12:0:
[   58s] /usr/include/stdlib.h:445:14: note: in a call to allocation function 'calloc' declared here
[   58s]  extern void *calloc (size_t __nmemb, size_t __size)
[   58s]               ^~~~~~
[   58s] /home/abuild/rpmbuild/BUILD/s25rttr-0.8.2/s-c/resample-1.8.1/sndlib-20/sound.c:919:8: error: argument 1 value '18446744073709551615' exceeds maximum object size 9223372036854775807 [-Werror=alloc-size-larger-than=]
[   58s]    samp = (mus_sample_t *)CALLOC(ichans, sizeof(mus_sample_t));
[   58s] In file included from /home/abuild/rpmbuild/BUILD/s25rttr-0.8.2/s-c/resample-1.8.1/sndlib-20/sound.c:12:0:
[   58s] /usr/include/stdlib.h:445:14: note: in a call to allocation function 'calloc' declared here
[   58s]  extern void *calloc (size_t __nmemb, size_t __size)
[   58s]               ^~~~~~
[   58s] [  5%] Building CXX object libsiedler2/src/CMakeFiles/siedler2.dir/ArchivItem_Bitmap_Player.cpp.o
[   58s] cc1: all warnings being treated as errors
[   58s] make[2]: *** [s-c/resample-1.8.1/sndlib-20/CMakeFiles/resample-sndlib-20.dir/build.make:87: s-c/resample-1.8.1/sndlib-20/CMakeFiles/resample-sndlib-20.dir/sound.c.o] Error 1
[   58s] make[2]: *** Waiting for unfinished jobs....
[   58s] [  5%] Building CXX object libsiedler2/src/CMakeFiles/siedler2.dir/ArchivItem_Bitmap_RLE.cpp.o
[   58s] [  5%] Building CXX object libutil/src/CMakeFiles/s25util.dir/LANDiscoveryService.cpp.o
[   58s] make[1]: *** [CMakeFiles/Makefile2:1083: s-c/resample-1.8.1/sndlib-20/CMakeFiles/resample-sndlib-20.dir/all] Error 2
[   58s] make[1]: *** Waiting for unfinished jobs....
```

**Comment by Flamefire** [comment]
Looks like a compiler bug to me. It seems it follows the path where ichans could be -1 but that can never happen

**Comment by Peuc** [comment]
The problem is back and prevents RTTR to be compiled in GCC 8.1:

```
In file included from /home/peuc/rttr-git/src/s25client/s-c/resample-1.8.1/sndlib-20/sound.c:36:
/home/peuc/rttr-git/src/s25client/s-c/resample-1.8.1/sndlib-20/sound.c: In function 'mus_sound_set_maxamps':
/home/peuc/rttr-git/src/s25client/s-c/resample-1.8.1/sndlib-20/_sndlib.h:399:22: error: argument 1 value '18446744073709551615' exceeds maximum object size 9223372036854775807 [-Werror=alloc-size-larger-than=]
 #define CALLOC(a, b) calloc((size_t)(a), (size_t)(b))
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/peuc/rttr-git/src/s25client/s-c/resample-1.8.1/sndlib-20/sound.c:1133:42: note: in expansion of macro 'CALLOC'
             sf->maxamps = (mus_sample_t*)CALLOC(ichans, sizeof(mus_sample_t));
                                          ^~~~~~
In file included from /home/peuc/rttr-git/src/s25client/s-c/resample-1.8.1/sndlib-20/sound.c:14:
/usr/include/stdlib.h:541:14: note: in a call to allocation function 'calloc' declared here
 extern void *calloc (size_t __nmemb, size_t __size)
              ^~~~~~
In file included from /home/peuc/rttr-git/src/s25client/s-c/resample-1.8.1/sndlib-20/sound.c:36:
/home/peuc/rttr-git/src/s25client/s-c/resample-1.8.1/sndlib-20/_sndlib.h:399:22: error: argument 1 value '18446744073709551615' exceeds maximum object size 9223372036854775807 [-Werror=alloc-size-larger-than=]
 #define CALLOC(a, b) calloc((size_t)(a), (size_t)(b))
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/peuc/rttr-git/src/s25client/s-c/resample-1.8.1/sndlib-20/sound.c:1134:36: note: in expansion of macro 'CALLOC'
             sf->maxtimes = (off_t*)CALLOC(ichans, sizeof(off_t));
                                    ^~~~~~
In file included from /home/peuc/rttr-git/src/s25client/s-c/resample-1.8.1/sndlib-20/sound.c:14:
/usr/include/stdlib.h:541:14: note: in a call to allocation function 'calloc' declared here
 extern void *calloc (size_t __nmemb, size_t __size)
              ^~~~~~
```

**Comment by Flamefire** [comment]
It would be great if anyone knows, how to avoid this. The problem is definitely that `ichans = -1` is propagated to the `calloc` call although it is checked earlier to be not -1. So at runtime this cannot happen and the warning is a false positive the compiler COULD recognize.

In the meantime use  `RTTR_ENABLE_WERROR=OFF` for CMake.

**Comment by Flamefire** [comment]
Ok, the compiler can't figure it out at that point. Fixing it...
However I did not get those warnings on my `GCC Ubuntu 8.1.0-3ubuntu2~16.04.york0`, so I could not fully test it. So I'll just push my fix and have you test it please.

**Comment by Flamefire** [comment]
Fixed with 5941f79338207aed4ac87aff4dcbfc5df5f79ade

**Comment by Peuc** [comment]
Looks like it's fixed. Thanks!
